### PR TITLE
scripts: nrfjprog: Use built-in pin reset enable

### PR DESCRIPTION
**Pull request by carlescufi**
Use nrfjprog's built-in pin reset enabling functionality instead of
doing it manually.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

### CODE REVIEW

File: scripts/support/runner/nrfjprog.py [hunk 1]
```diff
@@ -76,16 +76,11 @@ def do_run(self, command, **kwargs):
...
-                 '-f', self.family, '--snr', board_snr],
-                ['nrfjprog', '--reset', '-f', self.family, '--snr', board_snr],
+                # Enable pin reset
+                ['nrfjprog', '--pinresetenable', '-f', self.family,
+                 '--snr', board_snr],
             ])
-        commands.append(['nrfjprog',
-                         '--pinreset',
-                         '-f', self.family,
+        commands.append(['nrfjprog', '--pinreset', '-f', self.family,
```
Comment (mbolivar) [inline scripts/support/runner/nrfjprog.py]: Unrelated change ;)

**Review by mbolivar — APPROVED** [review]
Thanks! I ported those commands unchanged from the old shell scripts, didn't know about this option.

File: scripts/support/runner/nrfjprog.py (hunk 1 above)
Comment (carlescufi) [reply scripts/support/runner/nrfjprog.py]: sneaky... ;)
